feat(v6.4): update release notes for version 6.4 (#638)
invictus-integration/docs-ifa: versioned_docs/version-v6.0.0/support/release-notes.mdx

    - **[Transco] fall back on default parameter values:** the component better handles optional or unavailable data. In situations where the component can't retrieve a parameter value from a path, you can fall back on a `defaultValue`.

    - **Enhanced private network support:** deployments within a virtual network (VNET) offers private endpoints. This helps organizations to align their governance and standards in the deployment.
  </Features>
  <TechChanges>
    - **Upgrade to .NET 10:** the Dashboard and Framework components run on .NET 10.
  </TechChanges>
  <Fixes>
    - **[ExceptionHandler] report error details in failed `ForEach` loops:** the component captures inner errors inside `ForEach` loops in Logic Apps workflows instead of displaying general error user messages. Users no longer have to switch to Azure to find the root cause for failures.

>     - **Improved automatic recovery of deferred messages:** We have enhanced the automatic recovery mechanism for messages that remain in a deferred state for an extended period due to intermittent platform-related issues outside of Invictus' control.

Reviewer: ```suggestion
    - **[PubSub] Improved recovery of deferred messages:** the component recovers more reliably messages stuck in a deferred state, caused by external intermittent platform issues.
```

<ReleaseNotes>

<ReleaseVersion version="6.4" date="July 2026">
  <Features>

>     - **Simplified Flow Status Filtering:** We have streamlined the flow status filtering in the Dashboard Advanced Search to make it easier to find the flows you're looking for, while still maintaining detailed status information behind the scenes. Existing statuses such as Resubmitted, Resumed, Resubmit Failed, and Resume Failed are still tracked internally, but are now grouped into broader categories in the user interface. 

Reviewer: ```suggestion
    - **Simplify flow status filters into groups:** the flow status filters in the advanced search pane group statuses into broader categories instead of too detailed levels. 'Active' (= `Active`, `Resubmitted`, `Resumed` statuses). 'Error' (= `Failed`, `Resubmit Failed`, and `Resume Failed` statuses).
```
**requires change:** we're talking twice about the same thing, using 'now' three times. Think the sub-list makes it visually unnecessary heavy for the new set of statuses.

    
      The 'Active' filter now includes:
        - Active
        - Resubmitted
        - Resumed

      The 'Error' filter now includes:
        - Failed
        - Resubmit Failed
        - Resume Failed

> 

Reviewer: ```suggestion
```

>     - **Improved Handling of Missing Configuration References in Transco V2:** We've improved the way the Transco component handles configuration references that point to data that isn't available. Previously, these missing references would cause Transco to stop processing and return an error. This caused issues in scenarios where Transco was used as part of conditional logic that intentionally checks whether certain data exists before performing additional actions. With this enhancement, Transco will no longer fail when a referenced location is missing. Instead, it will continue processing and log a warning, making it easier to identify configuration issues while preserving existing conditional processing behavior.

Reviewer: ```suggestion
    - **[Transco] missing references no longer cause failures:** scenarios with conditional actions failed due to possible missing data references during instructions. The Transco no longer fails when `scopePath` or `destination` points to unavailable locations. The component logs a warning in case that happens to help spotting configuration issues.
```
**requires change:** we stopped using 'v2' but instead call v1 legacy API's and v2 container versions.

>     - **TranscoV2: Configurable Default Values for Missing Data:** The Transco component now provides more flexibility when working with optional or unavailable data. In situations where a configured value cannot be retrieved from the expected location, a configurable default value can now be used instead.

Reviewer: ```suggestion
    - **[Transco] fall back on default parameter values:** the component better handles optional or unavailable data. In situations where the component can't retrieve a parameter value from a path, you can fall back on a `defaultValue`.
```

>       This enhancement helps prevent processing failures caused by missing data and makes it easier to support scenarios where certain inputs are optional. The default value capability is available for both data lookups and context promotion operations, providing greater consistency and flexibility when configuring Transco behavior.

Reviewer: ```suggestion
```
**requires change:** seems like saying the same thing in another way.

>     - **Dedicated Resource Groups for Private Endpoints:** Invictus now provides additional flexibility when deploying environments that use Virtual Network (VNET) integration. Private Endpoints can now be deployed into a dedicated resource group, allowing organizations to align deployments more easily with their networking, governance, and resource management standards.

Reviewer: ```suggestion
    - **Enhanced private network support:** deployments within a virtual network (VNET) offers private endpoints. This helps organizations to align their governance and standards in the deployment.
```

**requires change:** not sure why this is considered a feature and the point below it, with private endpoints as a whole is considered technical change. Think they are both talking about the same thing as they are introduced at the same release time.

  </Features>
  <TechChanges>

>     - **Enhanced Private Network Deployment Support:** Invictus now offers an improved deployment model for environments that require private network connectivity. Deployments can now be configured to use Private Endpoints, ensuring that communication with underlying platform services stays within the private network boundary.

Reviewer: ```suggestion
```
See: https://github.com/invictus-integration/docs-ifa/pull/638/changes#r3557088612

>     We have also expanded deployment validation with automated connectivity checks to verify that services remain reachable from within the network while remaining inaccessible from external environments where appropriate.

Reviewer: ```suggestion
```
See: https://github.com/invictus-integration/docs-ifa/pull/638/changes#r3557088612

Not sure if we should mention how we test internally our infrastructure. Not really the place for that in the release notes, I think.

>     The update also simplifies the configuration of network-related deployment settings and improves support for managing private connectivity across the platform services used by Invictus.

Reviewer: ```suggestion
```
See: https://github.com/invictus-integration/docs-ifa/pull/638/changes#r3557088612

>     - **Upgrade to .NET 10:** We have upgraded All Dashboard and Framework functions from Invictus to .NET 10.

Reviewer: ```suggestion
    - **Upgrade to .NET 10:** the Dashboard and Framework components run on .NET 10.
```

  </TechChanges>
  <Fixes>
    - **ErrorHandler: Improved Error Details for failed ForEach Iterations:** We have improved the way errors are reported when a failure occurs inside a ForEach loop.
    Previously, if one or more iterations inside a ForEach action failed, the Dashboard would often display a generic error message such as "Unable to obtain error message" instead of the actual cause of the failure. This made troubleshooting more difficult and required users to inspect the Logic App run details directly.

>       With this enhancement, Invictus now retrieves and displays the underlying error from the failed iteration whenever possible, providing clearer and more actionable information in the Dashboard. This helps users identify the root cause of failures faster and reduces the need to switch to Azure for additional investigation.

Reviewer: ```suggestion
    - **[ExceptionHandler] report error details in failed `ForEach` loops:** the component captures inner errors inside `ForEach` loops in Logic Apps workflows instead of displaying general error user messages. Users no longer have to switch to Azure to find the root cause for failures.
```

    - **Improved automatic recovery of deferred messages:** We have enhanced the automatic recovery mechanism for messages that remain in a deferred state for an extended period due to intermittent platform-related issues outside of Invictus' control.

      Previously, when a deferred message was automatically resubmitted, environments with multiple subscribers could receive the message more broadly than intended. The subscription filtering logic has now been improved to ensure that automatically resubmitted messages are routed only to the appropriate subscriber.

>     - **Improved CosmosDB reliability during high-volume processing:** We've improved the way Invictus handles large numbers of background operations running at the same time. In certain scenarios, especially during initial setup or periods of high activity, this could result in database connection limitations that caused processing failures. The processing logic has been optimized to better manage concurrent operations, improving stability and reducing the likelihood of these errors occurring.

Reviewer: ```suggestion
    - **Reliable Cosmos DB during high-volume processing:** the database interactions better handles concurrent tasks. This reduces the connection errors during high activity or initial setup.
```
